content/articles/2021-08-01-dbee-nvim.md

```diff
@@ -0,0 +1,72 @@
+# My first vim plugin!
+
+Have you ever had the necessity of check a query while you are writing a python
+wrapper? If so, this post is for you.
+
+Many times I had the necessity of checking a query while I'm writing a function,
+debugging or testing it. That's why I wrote my first vim (well, actually a
+nvim) plugin called [`dbee.nvim`](https://github.com/mmngreco/dbee.nvim).
+
+## My using case
+
+I have a virtual environment with `SQLAlchemy` and others Python dependencies
+and then I want to check a query from my project. So, I have to open `DBeaver`
+or another SQL client copy-paste the query there and see the result. This
+process is very annoying.
+
+Why I have to open an external program when I already had all what I need in my
+virtual environment?
+
+The idea is define a `engine` using `SQLAlchemy` using a connection string, and
+then through `pandas` fetch the data using that engine. Pretty simple, isn't
+it?
+
+## Installation
+
+If you want to use it, add the following in your vimrc:
+
+```
+Plug 'mmngreco/dbee.nvim'
+```
+
+I'm using the following maps:
+
+```
+vnoremap <C-q> :DBeeQuery<cr>
+vnoremap <C-c><C-c> :DBeeSetConnection<cr><cr>
+```
+
+## Example
+
+Suppose we are writing the following python function:
+
+```python
+def get_titles():
+    cn_str = "sqlite:////home/mgreco/gitlab/mmngreco/dbee.nvim/tests/chinook.db"
+    engine = sa.create_engine(cn_str)
+    conn = engine.connect()
+    qry = """
+          SELECT title
+          FROM albums;
+          """
+    out = pd.read_sql(qry, conn)
+    return out
+```
+
+We can setup our connection by going to `cn_str` line, selecting the content
+inside the quotes and executing `:'<,'>DBeeSetConnection`. This will print out
+the following message: `Configurated
+sqlite:////home/mgreco/gitlab/mmngreco/dbee.nvim/tests/chinook.db`, after that
+we are ready to make queries to the database.
+
+Now, we are ready to select the query inside the quotes and call to
+`:'<,'>DBeeQuery`. This should open a new temporary buffer with the connection
+string, the query used and the output on it. You can easily close the buffer by
```

Comment:
string, the selected query and the output fetched from it. You can easily close the buffer by